test/_basic.jl

```diff
@@ -322,6 +322,21 @@ end
     @test axiskeys(push!(V2, 0)) === (Base.OneTo(4),)
     @test axiskeys(append!(V2, [7,7])) === (Base.OneTo(6),)
 
+    V3 = KeyedArray([3,5,7], [1, 5, 15])
+    # no natural value for integer axiskey:
+    @test_throws Exception push!(V3, 11)
+    @test_throws Exception append!(V3, [20, 25])
```

Comment:
I guess this case used to work? 

I presume that what I was thinking with `length+1` was so that `KeyedArray([pi, 2pi, 3pi], 1:3)` and `KeyedArray([pi, 2pi, 3pi], [1,2,3])` would have the same behaviour, i.e. accidentally `collect`ing doesn't change anything.

But maybe explicit is better.

Reply:
But `5:10` vs `collect(5:10)` would still give different results – there's simply no natural continuation for a `Vector`, no way around it...